src/gpl/src/mbff.cpp

```diff
@@ -0,0 +1,1219 @@
+        response.status() == operations_research::sat::CpSolverStatus::OPTIMAL) {
+
+        log_->info(utl::GPL, 22, "Total = {}", (response.objective_value()));
+
+        float sum_disp = 0;
+        for (int i = 0; i < num_flops; i++) {
+            for (int j = 0; j < num_trays; j++) {
+                for (int k = 0; k < static_cast<int>(trays[j].slots.size());
+                     k++) {
+                    if (trays[j].cand[k] == i) {
+                        sum_disp +=
+                            (0.01 *
+                             operations_research::sat::SolutionIntegerValue(
+                                 response, disp_x[i][j][k]));
+                        sum_disp +=
+                            (0.01 *
+                             operations_research::sat::SolutionIntegerValue(
+                                 response, disp_y[i][j][k]));
+                    }
+                }
+            }
+        }
+
+        log_->info(utl::GPL, 41, "Sum of displacements from flop to slot: {}",
+                   sum_disp);
+
+        float sum_disp_path = 0;
+        for (int i = 0; i < num_paths; i++) {
+            sum_disp_path +=
+                (0.01 * operations_research::sat::SolutionIntegerValue(
+                            response, disp_path_x[i]));
+            sum_disp_path +=
+                (0.01 * operations_research::sat::SolutionIntegerValue(
+                            response, disp_path_y[i]));
+        }
+
+        log_->info(utl::GPL, 412, "Sum of timing-critical path displacements: {}",
+                   sum_disp_path);
+
+        float sum_tray_cost = 0;
+        for (int i = 0; i < num_trays; i++) {
+            if (operations_research::sat::SolutionIntegerValue(response,
+                                                               tray_used[i])) {
+                sum_tray_cost += tray_cost[i];
+            }
+        }
+
+        log_->info(utl::GPL, 444, "Sum of tray costs: {}", sum_tray_cost);
+
+
+        std::set<int> tray_idx;
+        for (int i = 0; i < num_flops; i++) {
+            for (int j = 0; j < num_trays; j++) {
+                for (int k = 0; k < static_cast<int>(trays[j].slots.size());
+                     k++) {
+                    if (trays[j].cand[k] == i) {
+                        if (operations_research::sat::SolutionIntegerValue(
+                                response, mapped[i][j][k]) == 1) {
+                            tray_idx.insert(j);
+                        }
+                    }
+                }
+            }
+        }
+
+        std::map<int, int> tray_sz;
+        for (auto x : tray_idx) {
+            tray_sz[static_cast<int>(trays[x].slots.size())]++;
+        }
+
+        for (auto x : tray_sz) {
+            log_->error(utl::GPL, 512, "Tray size = {}: {}", x.first, x.second);
+        }
+
+    }
+
+} // end ILP
+
+std::vector<odb::Point> MBFF::GetSlots(const odb::Point &tray, int rows,
```

Comment:
Similar to the other comment. As already mentioned, you could use a reference here to avoid copying the whole vector.